src/nebula/NebulaAction.h

```diff
@@ -364,6 +406,107 @@ class CheckLeadersAction : public MetaAction {
     std::string spaceName_;
 };
 
+class CheckLeaderDistributionAction : public MetaAction {
+public:
+    CheckLeaderDistributionAction(GraphClient* client,
+                                  core::ActionContext* ctx,
+                                  const std::string& resultVarName,
+                                  const std::string& spaceName = "")
+        : MetaAction(client)
+        , ctx_(ctx)
+        , resultVarName_(resultVarName)
+        , spaceName_(spaceName) {}
+
+    ~CheckLeaderDistributionAction() = default;
+
+    ResultCode doRun() override;
+
+    ResultCode checkResp(const ExecutionResponse& resp);
+
+    std::string command() const override {
+        return "show hosts";
+    }
+
+private:
+    core::ActionContext*    ctx_{nullptr};
+    std::string             resultVarName_;
+    std::string             spaceName_;
+    std::string             restult_;
+};
+
+class SetFlagAction : public MetaAction {
+public:
+    SetFlagAction(GraphClient* client,
+                  const std::string& layer,
+                  const std::string& name,
+                  const std::string& value)
+        : MetaAction(client)
+        , layer_(layer)
+        , name_(name)
+        , value_(value) {}
+
+    ~SetFlagAction() = default;
+
+    ResultCode doRun() override;
+
+    ResultCode buildCmd() {
+        folly::toLowerAscii(layer_);
+        folly::toLowerAscii(name_);
+        folly::toLowerAscii(value_);
+        if (layer_ != "storage" && layer_ != "meta" && layer_ != "graph") {
+           return ResultCode::ERR_FAILED;
+        }
+
+        cmd_ ="update configs ";
+        cmd_ += layer_;
+        cmd_ += ":";
+        if (name_ == "disable_auto_compactions")  {
+            cmd_ += "rocksdb_column_family_options = { disable_auto_compactions = ";
+            if (value_ != "true" && value_ != "false") {
+                return ResultCode::ERR_FAILED;
+            }
+            cmd_ += value_;
+            cmd_ += " } ";
+        }
+        return ResultCode::OK;
+    }
+
+    std::string command() const override {
+        return cmd_;
+    }
+
+private:
+    std::string             layer_;
+    std::string             name_;
+    std::string             value_;
+    std::string             cmd_;
+};
+
+class CompareLeaderDistributionAction : public core::Action {
```

Comment:
How about just merge into `CheckLeaderAction`, add a flag to compare current result add previous result?